tests/trainer/test_checkpoint.py

```diff
@@ -915,6 +922,97 @@ def test_resumption(
             save_folder / 'second' / final_checkpoint,
         )
 
+    @pytest.mark.parametrize('world_size', [
+        pytest.param(2, marks=pytest.mark.world_size(2)),
+    ])
+    @pytest.mark.parametrize('device', [
+        pytest.param('gpu', marks=pytest.mark.gpu),
+    ])
+    @pytest.mark.parametrize(
+        'use_orig_params,sync_module_states,model_1_init_device,model_2_init_device',
+        [
+            pytest.param(False, True, 'cpu', 'cpu'),  # success
+            pytest.param(False, True, 'cpu', 'meta'),  # success
+            pytest.param(True, True, 'cpu', 'cpu'),  # fail
+            pytest.param(False, False, 'cpu', 'cpu'),  # fail
+            pytest.param(False, True, 'meta', 'cpu'),  # fail
+        ])
+    @pytest.mark.filterwarnings('ignore:An unexpected prefix is detected. This case.*')
+    @pytest.mark.filterwarnings(
+        'ignore:``FullyShardedDataParallel.scatter_full_optim_state_dict``is being deprecated and is replaced by.*')
+    def test_fsdp_monolith_resumption(
+        self,
+        device: str,
+        world_size: int,
+        use_orig_params: bool,
+        sync_module_states: bool,
+        model_1_init_device: str,
+        model_2_init_device: str,
+        tmp_path: pathlib.Path,
+    ):
+        save_interval = '1ba'
+        save_filename = 'ba{batch}-rank{rank}.pt'
+        resume_file = 'ba1-rank{rank}.pt'
+        final_checkpoint = 'latest-rank{rank}.pt'
+        fsdp_config = {
+            'use_orig_params': use_orig_params,
+            'sync_module_states': sync_module_states,
```

Comment:
do you want to explicitly set `state_dict_type='full'` here? In case we change a default later on?